File: examples/java/DynamoDbDeleteItem.java
// Elimina un item de DynamoDB por su clave primaria.
// Uso: java DynamoDbDeleteItem <tabla> <id>
import java.net.URI;
import java.util.Map;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
import software.amazon.awssdk.services.dynamodb.model.AttributeValue;

public class DynamoDbDeleteItem {
  public static void main(String[] args) {
    if (args.length < 2) {
      throw new IllegalArgumentException("Uso: java DynamoDbDeleteItem <tabla> <id>");
    }
    String tableName = args[0];
    String id = args[1];

    DynamoDbClient dynamodb = DynamoDbClient.builder()
        .endpointOverride(URI.create("http://localhost:4566"))
        .region(Region.US_EAST_1)
        .credentialsProvider(
            StaticCredentialsProvider.create(AwsBasicCredentials.create("test", "test")))
        .build();

    try {
      dynamodb.deleteItem(req -> req
          .tableName(tableName)
          .key(Map.of("id", AttributeValue.builder().s(id).build())));
      System.out.println("Item id=\"" + id + "\" eliminado de " + tableName + ".");
    } finally {
      dynamodb.close();
    }
  }
}

File: examples/python/dynamodb_delete_item.py
"""Elimina un item de DynamoDB por su clave primaria.

Uso: python3 dynamodb_delete_item.py <tabla> <id>
"""
import sys

import boto3

dynamodb = boto3.client(
    "dynamodb",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 3:
        raise SystemExit("Uso: python3 dynamodb_delete_item.py <tabla> <id>")
    table_name, item_id = sys.argv[1], sys.argv[2]

    dynamodb.delete_item(TableName=table_name, Key={"id": {"S": item_id}})
    print(f'Item id="{item_id}" eliminado de {table_name}.')


if __name__ == "__main__":
    main()
